components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/RoleBasedInternalScopeValidator.java

```diff
@@ -173,9 +200,16 @@ private List<String> getRolesOfTheUser(AuthenticatedUser authenticatedUser) thro
                     = (AbstractUserStoreManager) realmService.getTenantUserRealm(tenantId).getUserStoreManager();
 
             String userName = userStoreManager.getUserNameFromUserID(authenticatedUser.getUserId());
+            // If the username is empty try to set it as authenticated username.
+            if (StringUtils.isEmpty(userName)) {
+                userName = authenticatedUser.getUserName();
+            }
+            // Remove domain name from username.
+            if (StringUtils.isNotEmpty(userName)) {
+                userName = UserCoreUtil.removeDomainFromName(userName);
+            }
 
-            return userStoreManager.getHybridRoleListOfUser(UserCoreUtil.removeDomainFromName(userName),
-                    authenticatedUser.getUserStoreDomain());
+            return userStoreManager.getHybridRoleListOfUser(userName, authenticatedUser.getUserStoreDomain());
```

Comment:
check whether we can use ID here

Reply:
No interface for `AbstractUserStoreManager` based on ID for getHybridRoleListOfUser